torchao/quantization/quantize_/workflows/intx/intx_unpacked_tensor.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import List, Optional, Tuple
+
+import torch
+from torch.utils._python_dispatch import return_and_correct_aliasing
+
+from torchao.quantization.quant_primitives import (
+    _DTYPE_TO_BIT_WIDTH,
+    _DTYPE_TO_QVALUE_BOUNDS,
+    MappingType,
+    choose_qparams_affine,
+    dequantize_affine,
+    quantize_affine,
+)
+from torchao.utils import (
+    TORCH_VERSION_AT_LEAST_2_5,
+    TorchAOBaseTensor,
+    fill_defaults,
+)
+
+__all__ = [
+    "IntxUnpackedTensor",
+]
+
+aten = torch.ops.aten
+
+_FLOAT_TYPES: List[torch.dtype] = [torch.float16, torch.bfloat16, torch.float32]
+
+
+class IntxUnpackedTensor(TorchAOBaseTensor):
+    """
+    intx quantization with unpacked format.  Subbyte quantized data is represented as int8.
```

Comment:
nit: to make it clearer, we can add a bit more description here about subbyte quantized data I think, we should mention the range of the quantized values are restricted to the quant_min and quant_max of the target bit width, e.g. for uint4, the values falls into ranges of 0 and 15